packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataPersistenceService.java

```diff
@@ -32,20 +34,35 @@ public class RepresentationMetadataPersistenceService implements IRepresentation
 
     private final IRepresentationMetadataCreationService representationMetadataCreationService;
 
-    public RepresentationMetadataPersistenceService(IRepresentationMetadataCreationService representationMetadataCreationService) {
+    private final IRepresentationMetadataUpdateService representationMetadataUpdateService;
+
+    private final IRepresentationMetadataSearchService representationMetadataSearchService;
+
+    public RepresentationMetadataPersistenceService(IRepresentationMetadataCreationService representationMetadataCreationService,
+            IRepresentationMetadataUpdateService representationMetadataUpdateService,
+            IRepresentationMetadataSearchService representationMetadataSearchService) {
         this.representationMetadataCreationService = Objects.requireNonNull(representationMetadataCreationService);
+        this.representationMetadataUpdateService = Objects.requireNonNull(representationMetadataUpdateService);
+        this.representationMetadataSearchService = Objects.requireNonNull(representationMetadataSearchService);
     }
 
     @Override
     @Transactional
     public void save(ICause cause, IEditingContext editingContext, org.eclipse.sirius.components.core.RepresentationMetadata representationMetadata, String targetObjectId) {
         var optionalProjectId = new UUIDParser().parse(editingContext.getId());
-        var optionalRepresentationId = new UUIDParser().parse(representationMetadata.getId());
-        if (optionalProjectId.isPresent() && optionalRepresentationId.isPresent()) {
-            var projectId = optionalProjectId.get();
+        var optionalRepresentationId = new UUIDParser().parse(representationMetadata.id());
+
+        if (optionalRepresentationId.isPresent()) {
             var representationId = optionalRepresentationId.get();
-            this.representationMetadataCreationService.create(cause, representationId, projectId, representationMetadata.getLabel(), representationMetadata.getKind(),
-                    representationMetadata.getDescriptionId(), targetObjectId);
+
+            var exists = this.representationMetadataSearchService.existsById(representationId);
+            if (exists) {
+                this.representationMetadataUpdateService.updateRepresentationMetadataLabelById(cause, representationId, representationMetadata.label());
```

Comment:
This seems correct (at the moment), but still conceptually strange to unconditionally interpret "overwrite an existing representationmetadata" as "update its label".

Reply:
Oh ! Indeed ! That should be the update of the whole metadata